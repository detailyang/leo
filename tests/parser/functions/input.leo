/*
namespace: Parse
expectation: Pass
*/

function x(input) {
    return ();
}