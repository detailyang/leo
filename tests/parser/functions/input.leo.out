---
namespace: Parse
expectation: Pass
outputs:
  - name: ""
    expected_input: []
    imports: []
    circuits: {}
    functions:
      "{\"name\":\"x\",\"span\":\"{\\\"line_start\\\":3,\\\"line_stop\\\":3,\\\"col_start\\\":10,\\\"col_stop\\\":11,\\\"path\\\":\\\"input.leo\\\",\\\"content\\\":\\\"function x(input) {\\\"}\"}":
        annotations: []
        identifier: "{\"name\":\"x\",\"span\":\"{\\\"line_start\\\":3,\\\"line_stop\\\":3,\\\"col_start\\\":10,\\\"col_stop\\\":11,\\\"path\\\":\\\"input.leo\\\",\\\"content\\\":\\\"function x(input) {\\\"}\"}"
        input:
          - InputKeyword: "{\"name\":\"input\",\"span\":\"{\\\"line_start\\\":3,\\\"line_stop\\\":3,\\\"col_start\\\":12,\\\"col_stop\\\":17,\\\"path\\\":\\\"input.leo\\\",\\\"content\\\":\\\"function x(input) {\\\"}\"}"
        output: ~
        block:
          statements:
            - Return:
                expression:
                  TupleInit:
                    elements: []
                    span:
                      line_start: 4
                      line_stop: 4
                      col_start: 12
                      col_stop: 14
                      path: input.leo
                      content: "    return ();"
                span:
                  line_start: 4
                  line_stop: 4
                  col_start: 5
                  col_stop: 14
                  path: input.leo
                  content: "    return ();"
          span:
            line_start: 3
            line_stop: 5
            col_start: 19
            col_stop: 2
            path: input.leo
            content: "function x(input) {\n...\n}"
        span:
          line_start: 3
          line_stop: 5
          col_start: 1
          col_stop: 2
          path: input.leo
          content: "function x(input) {\n...\n}"
